### PR TITLE
KFC

### LINKED ISSUE #16: KFC
https://www.kfc.com/store-locator?query=90210

### PR DESCRIPTION
**Pull request by SelfnessAid**
Fixes #16

### CODE REVIEW

File: locations/spiders/kfc.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        yield scrapy.http.FormRequest(
+            url=url, method='POST', formdata=form_data,
+            headers=headers, callback=self.parse
+        )
+
+    def parse(self, response):
+        data = json.loads(response.body_as_unicode())
+        stores = data['results']
+        print(stores)
```
Comment (iandees) [inline locations/spiders/kfc.py]: Remove the `print()`

Comment (SelfnessAid) [reply locations/spiders/kfc.py]: ah~ sorry.
will do now.

**Comment by SelfnessAid** [comment]
Hi, @iandees 
One problem.
If we scrape the website with all zip code, it may occur timeout.
it's okay?

**Comment by iandees** [comment]
Yes, that's ok. A scraper can take as long as it needs to, but for the CI it only gets a minute to generate at least one result so that we can get feedback as quick as possible.

**Comment by SelfnessAid** [comment]
okay, got it.
thanks, @iandees

**Comment by SelfnessAid** [comment]
hi, @iandees 
please review my last update for #243 and let me know the result.
thanks.

File: locations/spiders/kfc.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                'ref': store['entityID'],
+                'name': store['storeNumber'],
+                'addr_full': store['addressLine'],
+                'city': store['city'],
+                'state': store['state'],
+                'postcode': store['postalCode'],
+                'lat': store['latitude'],
+                'lon': store['longitude'],
+                'phone': store['businessPhone'],
+                'opening_hours': 'Mo-Su 09:30-23:00'
```
Comment (iandees) [inline locations/spiders/kfc.py]: Surely some stores have different hours than this?

**Comment by iandees** [comment]
If you click on the "map" link in the scraperbot response you can see that there are lots of blank areas.

File: locations/spiders/kfc.py (hunk 2 above)
Comment (SelfnessAid) [reply locations/spiders/kfc.py]: hmm...
I think all stores on this website has same hours.
anyway, I'll fix it.
sorry, @iandees

**Comment by SelfnessAid** [comment]
what you mean blank area, @iandees ?

**Comment by iandees** [comment]
There are hundreds of KFC stores in Wisconsin, for example, and the scraper doesn't pick those up because your list of zipcodes isn't long enough.

**Comment by SelfnessAid** [comment]
okay, I'll add missing parts.

**Comment by iandees** [comment]
Cool, thanks! I'm working on a more efficient way of pulling out locations from APIs like this that do radius searches. Hopefully it is more efficient than just hitting a bunch of zip codes.

**Comment by SelfnessAid** [comment]
I think we can use uszipcode package to get all necessary zipcodes.

**Comment by SelfnessAid** [comment]
hi, @iandees 
I got 3252 items on my side and I am not sure why my result doesn't passed.
can you look at it ?
thanks.

**Comment by iandees** [comment]
If you click the "Log" link next to the word `exception` in [the bot comment above](https://github.com/alltheplaces/alltheplaces/pull/243#issuecomment-349420819), you can see what it failed on. It looks like one of the stores has an unexpected opening hours format.

**Comment by SelfnessAid** [comment]
well...
it works good on my side.
screen shot 2017-12-05 at 3 28 27 pm
@iandees I am not sure why result is different on my side and bot side.

File: locations/spiders/tacocabana.py [hunk 3]
```diff
@@ -1,36 +0,0 @@
-# -*- coding: utf-8 -*-
-import scrapy
-import json
-import re
-
-from locations.items import GeojsonPointItem
-
-class TacocabanaSpider(scrapy.Spider):
```
Comment (iandees) [inline locations/spiders/tacocabana.py]: Please don't remove this spider in your pull request.

**Pull request merged**